### PR TITLE
Fix/ensure secure connection

### PR DESCRIPTION
**Pull request by flora-hofmann-frequenz**
Continued work on #75

### CODE REVIEW

**Comment by Marenz** [comment]
LGTM, I would suggest you just merge your fixes into the first commit

**Comment by Marenz** [comment]
(btw, you can also force push to the original PR)

File: tests/test_client_reporting.py [hunk 1]
```diff
@@ -13,16 +13,16 @@
...
+    with patch("grpc.aio.secure_channel") as mock:
         yield mock
 
 
 @pytest.mark.asyncio
 async def test_client_initialization(mock_channel: MagicMock) -> None:
     """Test that the client initializes the channel."""
     client = ReportingApiClient("localhost:50051")  # noqa: F841
-    mock_channel.assert_called_once_with("localhost:50051")
+    mock_channel.assert_called_once_with("localhost:50051", ANY)
```
Comment (cwasicki) [inline tests/test_client_reporting.py:25]: Do you know why this is required for this change?

**Review by cwasicki — COMMENTED** [review]
LGTM, but agree that the commits should be squashed into one.

File: tests/test_client_reporting.py (hunk 1 above)
Comment (Marenz) [reply tests/test_client_reporting.py:25]: the ANY seems to mock the "key" parameter

**Review by cwasicki — APPROVED** [review]

**Review by sandovalrr — APPROVED** [review]
lgtm 🚀 . and passes all the unit tests  😆

**Pull request merged**